### ISSUE TITLE
Repeat section validation does not impact main form.

### ISSUE THREAD

**Issue by JimSangwine** [formly-js/ng-formly#384]
**I'm submitting a ...**

```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request
```

**Current behavior**

Repeat section FormGroups can be invalid, but they do not invalidate the parent FormGroup. I think this is because they do not have a parent assigned.

```
main_form_group: {
    _status: 'VALID',
    controls: {
        my_repeat_section: { // FormArray
            _status: 'VALID',
            _touched: false,
            parent: { /* FormGroup */ },
            controls: [
                { // FormGroup
                    _status: 'INVALID',
                    _touched: true,
                    parent: undefined
                }
            ]
        }
    }
}
```

**Expected behavior**

When there are validation errors in a repeat section the whole form should be invalid.

**Minimal reproduction of the problem with instructions**

https://plnkr.co/edit/m8R0G0xntyinA2Ngi9KD?p=preview 

Add a repeat section for which one or more fields is required to trigger the validation.

**What is the motivation / use case for changing the behavior?**

To enable consistent validation.

**Please tell us about your environment:**

OSX, npm

* **Angular version:** 

2.4.6

* **Browser:** 
[ Chrome | Firefox | Safari  ]

* **Language:** 
[ TypeScript 2.1.5 ]

**Comment by aitboudad** [comment]
could you provide a plunkr or a repo reproducing the issue? 
Not sure if it's related to `ng-formly` or `@angular/forms`

**Comment by JimSangwine** [comment]
Sure, I've updated the original post with a plunkr.

This is using your demo plunkr (so old versions of everything) but the behaviour is identical on the versions I specified.

**Comment by franzeal** [comment]
Try this:
https://plnkr.co/edit/8UikbtwdRMqVr8cQPLox?p=preview

**Comment by JimSangwine** [comment]
Thanks @franzeal - that seems to work.

For future visitors, the magic is in the repeatedSection file.

**Comment by aitboudad** [comment]
I see what's happen now, instead of using `ChangeDetectorRef` I would using `updateValueAndValidity` but still a workaround to me :)

```ts
Observable.timer().subscribe(() => (<FormArray>this.formControl).updateValueAndValidity());
```

once I have some time (maybe next week) I'll try to find a clean way

**Comment by JimSangwine** [comment]
Thanks very much guys!

**Comment by aitboudad** [comment]
fixed by #427
